### DISCUSSION TITLE
The first step within YouTube transcribing should be done client side.

### DISCUSSION THREAD

**Discussion by arjunkrishna** [open-webui/open-webui#1959] [Ideas]
The first step in YouTube transcribing where we add the url after #, does it fetch the transcript from YouTube on client side or the processing of download of the script happens on the server side?

**Comment by arjunkrishna** [comment]
it seems like even the initial download of the transcript for youtube video is happening on server side. it would be desirable if the transcript was downloaded on the client side and then rag api for youtube is called with the transcript text and url of the youtube video. this way the actual server running open-webui can remain disconnected from the internet.

**Reply by hoohooprog** [reply]
Have u done it yet? 😂

**Reply by arjunkrishna** [reply]
ended up going with copilot licenses for business users. was testing various offerings online.
